torch/csrc/lazy/ts_backend/ts_backend_impl.cpp

```diff
@@ -43,6 +46,19 @@ class TSBackendImpl : public torch::lazy::BackendImplInterface {
     auto type = (env_use_cuda || FLAGS_torch_lazy_ts_cuda) ? at::kCUDA : at::kCPU;
     default_device_type_ = TSBackendDeviceType(type);
   }
+
+  Shape GetShapeFromFunction(std::function<Shape()> shape_fn, hash_t hash) const override {
+    if (!hash) {
```

Comment:
i guess technically this isn't valid, but you're assuming a hash value of 0 is unlikely? When do we call with hash=0?

Reply:
i see; maybe make it c10::optional<hash_t> and provide nullopt as default instead just to be correct

Reply:
hmm. That is a good point. I added the check because I set the default value to 0 in the parent `BackendImplInterface`. But maybe the default value isn't needed